todo_app/app.py

```diff
@@ -152,8 +152,11 @@ def index():
     @login_required
     def create():
         print(current_user.id)
-        title = request.form.get('title')
-        create_items(title)
+        if current_user.role == "writer":
+            title = request.form.get('title')
+            create_items(title)
+        else:
+            print('User does not have writer role.')
```

Comment:
This is the right sort of idea, but note that:
- You have only secured creating items. Anyone can still modify items.
- It would be nice if your response makes it clear that you refused their request. Either a message on the page, or an error with status code 401 and a message that explains the problem.

Reply:
Updated, thank you